tests/cases/compiler/deferredConditionalTypes.ts

```diff
@@ -18,3 +18,21 @@ type Extends<A, B> = A extends B ? true : false;
 type IsNumberLiteral<T> = And<Extends<T, number>, Not<Extends<number, T>>>;
 
 type IsLiteral<T> = Or<false, IsNumberLiteral<T>>;
+
+// Repro from #51145#issuecomment-1276804047

```

Comment:
note that this test case **doesn't** showcase that #51145 has been fixed - it's just [a variant proposed by @RyanCavanaugh](https://github.com/microsoft/TypeScript/issues/51145#issuecomment-1276804047) there that didn't work either